cartographer_grpc/io/inmemory_proto_stream.cc

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer_grpc/io/inmemory_proto_stream.h"
+
+namespace cartographer_grpc {
+namespace io {
+
+bool InmemoryProtoStreamReader::ReadProto(google::protobuf::Message* proto) {
+  if (eof()) return false;
+
+  switch (load_map_requests_.front().map_chunk_case()) {
```

Comment:
Let's make this more general and not talk about the concrete types here. Use a queue<unqiue_ptr<Message>>.